AElf.OS.Network.Grpc/GrpcNetworkManager.cs

```diff
@@ -0,0 +1,275 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel;
+using AElf.OS.Network.Grpc.Events;
+using AElf.OS.Network.Temp;
+using Google.Protobuf;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus.Local;
+
+namespace AElf.OS.Network.Grpc
+{
+    public class GrpcNetworkManager : INetworkManager, IPeerAuthentificator, ISingletonDependency
+    {
+        private readonly IAccountService _accountService;
+        private readonly IBlockService _blockService;
+        private readonly ILocalEventBus _localEventBus;
+        
+        public ILogger<GrpcNetworkManager> Logger { get; set; }
+                
+        private readonly NetworkOptions _networkOptions;
+
+        private Server _server;
+
+        private List<GrpcPeer> _authenticatedPeers;
+        
+        public GrpcNetworkManager(IOptionsSnapshot<NetworkOptions> options, 
+            IAccountService accountService, IBlockService blockService, ILocalEventBus localEventBus)
+        {
+            _accountService = accountService;
+            _blockService = blockService;
+            Logger = NullLogger<GrpcNetworkManager>.Instance;
+            
+             _localEventBus = localEventBus;
+            
+            _authenticatedPeers = new List<GrpcPeer>();
+            
+            _networkOptions = options.Value;
+        }
+        
+        public async Task StartAsync()
```

Comment:
a manager should not have status, so we need to remove server start out of manager

we should make the server as singleton, and start it in the OnApplicationInitialize method in the module of the project

Reply:
I agree, and in this case I propose we rename the class and not use a "Manager" for the network layer (I don't see why we should use one actually (?)). This class (I'm not sure what to name it yet) can of course be Singleton, no problem.
Also ok for using OnApplicationInit for starting the net layer.